pkg/operator/staticpodoperator/controlloops/installer_controller.go

```diff
@@ -0,0 +1,377 @@
+	if err != nil {
+		return err
+	}
+	operatorStatus := originalOperatorStatus.DeepCopy()
+
+	switch operatorSpec.ManagementState {
+	case operatorv1alpha1.Unmanaged:
+		return nil
+	case operatorv1alpha1.Removed:
+		// TODO probably just fail.  Static pod managers can't be removed.
+		return nil
+	}
+
+	requeue, syncErr := c.createInstallerController(operatorSpec, operatorStatus, resourceVersion)
+	if requeue && syncErr == nil {
+		return fmt.Errorf("synthetic requeue request")
+	}
+	err = syncErr
+
+	if err != nil {
+		if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
+			v1alpha1helpers.SetOperatorCondition(&operatorStatus.Conditions, operatorv1alpha1.OperatorCondition{
+				Type:    operatorv1alpha1.OperatorStatusTypeFailing,
+				Status:  operatorv1alpha1.ConditionTrue,
+				Reason:  "StatusUpdateError",
+				Message: err.Error(),
+			})
+			if _, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus); updateError != nil {
+				glog.Error(updateError)
+			}
+		}
+		return err
+	}
+
+	return nil
+}
+
+// Run starts the kube-apiserver and blocks until stopCh is closed.
+func (c *InstallerController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	glog.Infof("Starting InstallerController")
+	defer glog.Infof("Shutting down InstallerController")
+
+	// doesn't matter what workers say, only start one.
+	go wait.Until(c.runWorker, time.Second, stopCh)
+
+	<-stopCh
+}
+
+func (c *InstallerController) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+func (c *InstallerController) processNextWorkItem() bool {
+	dsKey, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(dsKey)
+
+	err := c.sync()
+	if err == nil {
+		c.queue.Forget(dsKey)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %v", dsKey, err))
+	c.queue.AddRateLimited(dsKey)
+
+	return true
+}
+
+// eventHandler queues the operator to check spec and status
+func (c *InstallerController) eventHandler() cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.queue.Add(installerControllerWorkQueueKey) },
+		UpdateFunc: func(old, new interface{}) { c.queue.Add(installerControllerWorkQueueKey) },
+		DeleteFunc: func(obj interface{}) { c.queue.Add(installerControllerWorkQueueKey) },
+	}
+}
+
+const installerPod = `apiVersion: v1
+kind: Pod
+metadata:
+  namespace: openshift-kube-apiserver
+  name: installer-<deployment-id>-<nodeName>
+  labels:
+    app: installer
+spec:
+  serviceAccountName: installer-sa
+  containers:
+  - name: apiserver
+    image: ${IMAGE}
+    imagePullPolicy: Always
+    command: ["cluster-kube-apiserver-operator", "installer"]
+    securityContext:
+      privileged: true
+      runAsUser: 0
+    terminationMessagePolicy: FallbackToLogsOnError
+    volumeMounts:
+    - mountPath: /etc/kubernetes/
+      name: kubelet-dir
+  restartPolicy: Never
```

Comment:
don't we want to restart this on errors?

Reply:
> don't we want to restart this on errors?

I don't think so.  I think we need to have a definitive termination to know it failed.